zebra-consensus/src/block.rs

```diff
@@ -91,7 +91,17 @@ where
 
             // Check that this block is actually a new block
             if BlockVerifier::get_block(&mut state, hash).await?.is_some() {
-                Err(format!("Block has already been verified. {:?} {:?}", height, hash))?;
+                // Treat duplicate blocks as a harmless mistake.
+                //
+                // Otherwise, the syncer restarts twice on errors, once on the
+                // original error, and again as it re-submits duplicate blocks.
+                //
+                // TODO:
+                //   - use an error kind enum
+                //   - modify the syncer so it doesn't restart on duplicate
+                //     verify errors
+                tracing::info!(height = ?height, hash = ?hash, "Block is already in the state");
+                return Ok(hash);
             }
```

Comment:
If the syncer is operating correctly, and we have good peers, then this code should be unreachable, because:
* the block locator starts at the state tip
* peers should extend the block locator tip forwards

If we have bad peers, then we want to restart. If we have peers that aren't synced yet, then we might want to restart to pick up missing blocks.

So I don't think this change is needed.